src/lib/staking.ts

```diff
@@ -0,0 +1,171 @@
+import * as ERC20ABI from './abi/erc20.abi.json';
+import * as BVaultABI from './abi/balancer-vault.abi.json';
+import * as SPSPABI from './abi/spsp.abi.json';
+import * as MultiCallerABI from './abi/multicaller.abi.json';
+import { Contract } from 'ethers';
+import { PoolConfigsMap } from './pool-info';
+import {
+  BalancerVaultAddress,
+  Balancer_80PSP_20WETH_address,
+  CHAIN_ID_MAINNET,
+  MULTICALL_ADDRESS,
+  SAFETY_MODULE_ADDRESS,
+  PSP_ADDRESS,
+  MulticallEncodedData,
+  Balancer_80PSP_20WETH_poolId,
+} from './constants';
+import { Provider } from './provider';
+import { Interface } from '@ethersproject/abi';
+
+export type Stakes<T> = {
+  totalPSPStaked: T;
+  descr: {
+    totalPSPStakedInSPSP: T;
+    totalPSPStakedInSafetyModule: T;
+  };
+};
+
+export const SPSPAddresses = PoolConfigsMap[CHAIN_ID_MAINNET].filter(
+  p => p.isActive,
+).map(p => p.address.toLowerCase());
```

Comment:
we make this filtering elsewhere I think, perhaps an `activeSPSPAddresses` const in `./pool-info` could be usedful.